packages/gateway/src/utils/cid.js

```diff
@@ -0,0 +1,34 @@
+import { CID } from 'multiformats/cid'
+
+import { InvalidIpfsPathError, InvalidUrlError } from '../errors.js'
+
+/**
+ * Parse subdomain URL and return cid
+ *
+ * @param {string} url
+ */
+export function getCidFromSubdomainUrl(url) {
+  // Replace "ipfs-staging" by "ipfs" if needed
+  const nUrl = url.replace('ipfs-staging', 'ipfs')
+  const splitUrl = nUrl.split('.ipfs.')
```

Comment:
At this point is `splitUrl` not `['https://bafy...', 'nft.storage']`? `splitUrl[0]` will fail to pass as a CID.

Reply:
Ok, the parameter name was unfortunate, it is the hostname:

```js
getCidFromSubdomainUrl(url.hostname)
```

So, I guess just rename the parameter should be enough